### PR TITLE
TOOLS-3610 Add build.go targets for adding and updating dependencies

### PR DESCRIPTION
**Pull request by autarch**
This adds several new commands to the `build.go` file. These handle adding or updating deps, and will also tidy our `go.{mod,sum}` files, vendor the dependency, and update both the SBOM Lite and `THIRD-PARTY-NOTICES` files.

Most of this code was copied from the mongosync repo and then adjusted for `goke`.

### CODE REVIEW

File: buildscript/deps.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	if err == nil {
+		return nil
+	}
+
+	fmt.Println(`This command requires the "podman" CLI tool, which you will need to install.`)
+	fmt.Println("See https://podman.io/ for more information and installation instructions.")
+	return err
+}
+
+//nolint:misspell // "license" is intentional here
```
Comment (mmcclimon) [inline buildscript/deps.go:50]:
```suggestion
//nolint:misspell // "licence" is intentional here
```

(I think.)

File: common.yml [hunk 2]
```diff
@@ -532,17 +532,6 @@ functions:
...
-  "check third-party notices":
-    - command: shell.exec
-      type: test
-      params:
-        working_dir: src/github.com/mongodb/mongo-tools
-        script: |
-          set -e
-          set -x
-          ./etc/generate-notices.pl > ./THIRD-PARTY-NOTICES
-          git diff --exit-code
```
Comment (mmcclimon) [inline common.yml:544]: Before this checked that the notices were correct, and now it only writes them; is that what we want? (It seems like not.)

File: etc/generate-notices.pl [hunk 3]
```diff
@@ -1,438 +0,0 @@
-#!/usr/bin/env perl
```
Comment (mmcclimon) [inline etc/generate-notices.pl:1]: RIP to the last perl remaining in the repo.

File: vendor/github.com/deckarep/golang-set/v2/new_improved.jpeg
Comment (mmcclimon) [inline vendor/github.com/deckarep/golang-set/v2/new_improved.jpeg:1]: (It's so annoying that this dang jpeg file gets included.)

**Review by mmcclimon — APPROVED** [review]
I had one comment which is more of a question, but LGTM other than that.

File: common.yml (hunk 2 above)
Comment (autarch) [reply common.yml:544]: Oops, good catch! Yes, it should check for a diff. Fixed.

File: etc/generate-notices.pl (hunk 3 above)
Comment (autarch) [reply etc/generate-notices.pl:1]: :moment-of-silence:

**Review by tdq45gj — COMMENTED** [review]
It looks good. I just wanted to check if there's documentation on how to generate the SBOM lite file.

File: build.go [hunk 4]
```diff
@@ -16,8 +16,13 @@ func init() {
 	taskRegistry.Declare("build").Description("build the tools").OptionalArgs("pkgs").Do(buildscript.BuildTools)
 	taskRegistry.Declare("checkMinVersion").Description("check if the minimum required Go version exists").Do(buildscript.CheckMinimumGoVersion)
 
-	// SSDLC
-	taskRegistry.Declare("sbom:write").Description("create an SBOM Lite file using the Silkbomb tool").Do(buildscript.WriteSBOMLite)
+	// Deps & SSDL
+	taskRegistry.Declare("writeSBOMLite").Description("create an SBOM Lite file using the Silkbomb tool").Do(buildscript.WriteSBOMLite)
```
[the remark this answers is not in the document]
Comment (autarch) [reply build.go:20]: Oops, good catch. The code should check if `podman` is installed before trying to run it. If it's not, it points you at the install instructions on the podman site.

[the remark this answers is not in the document]
Comment (tdq45gj) [reply build.go:20]: I think I do have `podman` installed. This error is happening after `requirePodman`.

[the remark this answers is not in the document]
Comment (autarch) [reply build.go:20]: I _just_ pushed the change to actually check for podman in `WriteSBOMLite`. But if this is still happening and you have podman installed then there's something wrong with how it's set up.

[the remark this answers is not in the document]
Comment (mmcclimon) [reply build.go:20]: It sounds like you have podman _installed_, Jian, but that it's not running. Did you try the suggested commands (podman machine init, podman machine start)? It stands up a persistent small VM or something and runs things inside that container, so it's not enough that it's just installed.

[the remark this answers is not in the document]
Comment (tdq45gj) [reply build.go:20]: I did check out the latest change and did some searches online. Running `podman machine init` and `podman machine start` solved this error. Should we add them to the script?

[the remark this answers is not in the document]
Comment (autarch) [reply build.go:20]: What exactly would we check for in this case? I think this is something you need to do on macOS but not Linux, so I can't really test this scenario.

[the remark this answers is not in the document]
Comment (mmcclimon) [reply build.go:20]: Yeah, I think there's not really anything useful to do to our code here. The error message you get when this happens is fairly descriptive (it tells you the commands to run), so I think it's fine without. It's the kind of thing you're likely to see once and then remember "oh yeah, that thing," which I think is acceptable for something like this where you're going to do it pretty infrequently _anyway_.

[the remark this answers is not in the document]
Comment (tdq45gj) [reply build.go:20]: I have some changes to automate this process and have tested locally. Can I push it to this PR?

[the remark this answers is not in the document]
Comment (autarch) [reply build.go:20]: @tdq45gj Can you push a branch so I can take a look? Then I'll merge it into this PR's branch.

[the remark this answers is not in the document]
Comment (tdq45gj) [reply build.go:20]: Sure, it's #715

[the remark this answers is not in the document]
Comment (autarch) [reply build.go:20]: Merged into this branch.

**Review by tdq45gj — APPROVED** [review]
LGTM!

**Pull request merged**